vmm/src/memory_manager.rs

```diff
@@ -1845,6 +1849,38 @@ impl MemoryManager {
     pub fn acpi_address(&self) -> Option<GuestAddress> {
         self.acpi_address
     }
+
+    pub fn get_mapping_num(&self) -> u32 {
+        self.guest_ram_mappings.len() as u32
+    }
+
+    #[cfg(feature = "guest_debug")]
+    pub fn coredump_data(&self, mem_offset: u64) -> MemoryCoredumpData {
+        let mut mem_blocks = BTreeMap::new();
+        let mut offset = mem_offset;
+        for guest_ram_mapping in &self.guest_ram_mappings {
+            let load_info = MemCoredumpFileInfo {
+                offset: 0,
+                filesz: 0,
+                size: guest_ram_mapping.size,
+            };
+            mem_blocks.insert(guest_ram_mapping.gpa, load_info);
+        }
+
+        let mut ram_maps = BTreeMap::new();
+        for (gpa, load) in &mem_blocks {
+            let load_info = MemCoredumpFileInfo {
+                offset,
+                filesz: gpa + load.size,
+                size: load.size,
+            };
+
+            ram_maps.insert(*gpa, load_info);
+            offset += load.size;
+        }
```

Comment:
Can't you simplify this whole block by going through the list of `guest_ram_mappings` only one time?